consensus/types/src/data_column_sidecar.rs

```diff
@@ -62,94 +63,84 @@ pub struct DataColumnSidecar<T: EthSpec> {
-        let num_of_blobs = blob_sidecars.len();
-
-        (0..T::number_of_columns())
-            .map(|col_index| {
-                Ok(DataColumnSidecar {
-                    index: col_index as u64,
-                    column: Self::generate_column_data(&mut rng, num_of_blobs, col_index)?,
-                    kzg_commitments: blob_sidecars
-                        .iter()
-                        .map(|b| b.kzg_commitment)
-                        .collect::<Vec<_>>()
-                        .into(),
-                    kzg_proofs: blob_sidecars
-                        .iter()
-                        .map(|b| b.kzg_proof)
-                        .collect::<Vec<_>>()
-                        .into(),
-                    signed_block_header: first_blob_sidecar.signed_block_header.clone(),
-                    kzg_commitments_inclusion_proof: kzg_commitments_inclusion_proof.clone(),
-                })
-            })
-            .collect::<Result<Vec<_>, _>>()
+    pub fn slot(&self) -> Slot {
+        self.signed_block_header.message.slot
     }
 
-    fn generate_column_data(
-        rng: &mut StdRng,
-        num_of_blobs: usize,
-        index: usize,
-    ) -> Result<DataColumn<T>, DataColumnSidecarError> {
-        let mut dummy_cell_data = Cell::<T>::default();
-        // Prefix with column index
-        let prefix = index.to_le_bytes();
-        dummy_cell_data
-            .get_mut(..prefix.len())
-            .ok_or(DataColumnSidecarError::DataColumnIndexOutOfBounds)?
-            .copy_from_slice(&prefix);
-        // Fill the rest of the vec with random values
-        rng.fill(
-            dummy_cell_data
-                .get_mut(prefix.len()..)
-                .ok_or(DataColumnSidecarError::DataColumnIndexOutOfBounds)?,
-        );
-
-        let column = DataColumn::<T>::new(vec![dummy_cell_data; num_of_blobs])?;
-        Ok(column)
+    pub fn block_root(&self) -> Hash256 {
+        self.signed_block_header.message.tree_hash_root()
     }
 
-    pub fn id(&self) -> DataColumnIdentifier {
-        DataColumnIdentifier {
-            block_root: self.block_root(),
-            index: self.index,
+    pub fn build_sidecars(
+        blobs: &BlobSidecarList<T>,
+        block: &SignedBeaconBlock<T>,
+        kzg: &Kzg,
+    ) -> Result<DataColumnSidecarList<T>, DataColumnSidecarError> {
+        let kzg_commitments = block
+            .message()
+            .body()
+            .blob_kzg_commitments()
+            .map_err(|_err| DataColumnSidecarError::PreDeneb)?;
+        let kzg_commitments_inclusion_proof =
+            block.message().body().kzg_commitments_merkle_proof()?;
+        let signed_block_header = block.signed_block_header();
+
+        let mut columns =
+            vec![Vec::with_capacity(T::max_blobs_per_block()); T::number_of_columns()];
+        let mut column_kzg_proofs =
+            vec![Vec::with_capacity(T::max_blob_commitments_per_block()); T::number_of_columns()];
```

Comment:
I'm not sure if we actually need to allocate `max_blob_commitments_per_block` for this vec here. It's used for the `VariableList` max length (which is used for merkeization and impacts the root), but i don't think the capacity of the underlying vec has an impact on SSZ.

Reply:
Updated in https://github.com/sigp/lighthouse/pull/5210/commits/75cf05f3c746fccf7fbf9c7583a760ff60e72adf
(please correct me if I'm wrong though)